WorkAndFun/src/test/java/pl/infoshare/workandfun/announcements/AnnouncementServiceTest.java

```diff
@@ -162,4 +150,108 @@ void shouldThrowExceptionWhenNotFoundAnnouncementToDeleteById() {
+                false, "Andrzej, czyli ja to miłośnik zwierząt chętnie spędzający z nimi czas, nie masz co zrobić ze swoim zwierzakiem, zadzwoń do Andrzeja",
+                "+48666666666", "z FV będzie drożej");
+        given(addAndEditMapper.toEntity(dto)).willCallRealMethod();
+        given(announcementsRepository.save(captor.capture())).willReturn(announcement);
+        //when
+        announcementService.save(dto);
+        //then
+        verify(addAndEditMapper).toEntity(dto);
+        verify(announcementsRepository).save(any(Announcement.class));
+    }
+
+    @Test
+    void shouldUpdate() {
+        //given
+        AddAndEditAnnouncementDto dto = new AddAndEditAnnouncementDto(1L, Type.SERVICE_OFFER, "Wyprowadzam psy, koty, myszy, konie, słonie",
+                ServiceType.INNE, "Warszawa", "dzielnica", "osiedle", "200", Voivodeship.MAZOWIECKIE, "Andrzej", "andrzej@aa.pl",
+                false, "Andrzej, czyli ja to miłośnik zwierząt chętnie spędzający z nimi czas, nie masz co zrobić ze swoim zwierzakiem, zadzwoń do Andrzeja",
+                "+48666666666", "z FV będzie drożej");
+        given(announcementsRepository.findById(id)).willReturn(Optional.of(announcement));
+        given(announcementsRepository.save(announcement)).willReturn(announcement);
+        //when
+        Announcement actualResult = announcementService.update(id, dto);
+        //then
+        assertThat(actualResult).isSameAs(announcement);
+        verify(announcementsRepository).findById(id);
+        verify(announcementsRepository).save(same(announcement));
+    }
+
+    @Test
+    void shouldSearchAllByParameterNonBlank() {
+        //given
+        List<QuickViewAnnouncementDto> quickViewAnnouncementDtoList = new ArrayList<>(List.of(
+                new QuickViewAnnouncementDto(1L, "Wyprowadzam psy, koty, myszy, konie, słonie",
+                        "200", false, LocalDateTime.of(2021, 12, 20, 10, 15), false, "Wrocław, Bielany",
+                        "Andrzej, czyli ja to miłośnik zwierząt chętnie spędzający z nimi czas, nie masz co zrobić ze swoim zwierzakiem, zadzwoń do Andrzeja"),
+                new QuickViewAnnouncementDto(2L, "Mechanik samochodowy",
+                        "do ustalenia indywidualnie", true, LocalDateTime.of(2020, 10, 10, 10, 10), true, "Warszawa, Wola",
+                        "Diagnostyka, wymiana filtrów/oleju, pompowanie kół"),
+                new QuickViewAnnouncementDto(3L, "Przeprowadzki",
+                        "150", false, LocalDateTime.of(2022, 1, 10, 11, 25), false, "Kraków",
+                        "Mam bus, pomogę z przeprowadzką")));
+        doReturn(quickViewAnnouncementDtoList).when(announcementService).findAllSortedByCreateDateDescConvertToDto();
+//        given(announcementService.findAllSortedByCreateDateDescConvertToDto()).willReturn(quickViewAnnouncementDtoList);
```

Comment:
"//" w zlym miejscu

Reply:
oj nie, tu akurat w dobrym. Linijki teoretycznie robią to samo, tyle tylko że druga nie działa całkowicie, i wywala bład. Jest taki problem przy Spy-jach, że może nie działać given, a będzie działać tylko doReturn. Długo nie rozumiałem dlaczego nie działa, aż w końcu gdzieś natrafiłem na taką uwagę w dokumentacji.
Do usunięcia linijkę z zakomentowaną